src/EditorFeatures/Core/FindUsages/AbstractFindUsagesService.cs

```diff
@@ -63,26 +64,31 @@ protected AbstractFindUsagesService(IThreadingContext threadingContext)
         {
             var definitionTrackingContext = new DefinitionTrackingContext(context);
 
-            // Need ConfigureAwait(true) here so we get back to the UI thread before calling 
+            // Need ConfigureAwait(true) here so we get back to the UI thread before calling
             // GetThirdPartyDefinitions.  We need to call that on the UI thread to match behavior
             // of how the language service always worked in the past.
             //
             // Any async calls before GetThirdPartyDefinitions must be ConfigureAwait(true).
             await FindLiteralOrSymbolReferencesAsync(
                 document, position, definitionTrackingContext).ConfigureAwait(true);
 
-            // After the FAR engine is done call into any third party extensions to see
-            // if they want to add results.
-            var thirdPartyDefinitions = GetThirdPartyDefinitions(
+            // When editor handles symbol search, we are running on background thread.
+            // Don't call into third party extensions which work on UI thread
+            if (!SymbolSearchPreviewUtility.EditorHandlesSymbolSearch(document.Project.Solution.Workspace))
```

Comment:
fwiw I think I need something similar for liveshare / nexus to avoid the UI thread requirement.  should we create a IFindUsages2 service or something similar with version of ```FindReferencesAsync``` that doesn't call into third party references (and doesn't work on the UI thread)?
cc @jasonmalinowski for thoughts here as well

Reply:
does this break 3rd parties?

Reply:
_At the places that call into this from the new API entrypoints_, can we just SwitchToMainThreadAsync() with JTF? Or can we just break out the GetThirdPartyDefinitions() call to something separate that it deals with that way?